### ISSUE TITLE
Fase 2: Arquitectura del “Córtex Visual” (Construcción del Modelo)

### ISSUE THREAD

**Issue by Serdan1** [Serdan1/cnn-cifar10#2]
Descripción:
En esta fase se construirá la arquitectura base de la red neuronal convolucional (CNN) utilizando TensorFlow/Keras.
El objetivo es crear un modelo que combine un extractor de características (capas convolucionales y de pooling) con un clasificador (capas densas), simulando el funcionamiento del córtex visual humano al procesar imágenes.

🎯 Objetivos

Iniciar un modelo secuencial (tf.keras.Sequential()).

Implementar dos bloques convolucionales para extraer características visuales.

Construir un clasificador completamente conectado.

Preparar el modelo para la fase de entrenamiento.

Responder la Pregunta de Análisis 1 sobre la correspondencia entre el modelo y el flujo de visión artificial tradicional.

⚙️ Tareas

 Crear el modelo con Sequential().

 Añadir el Bloque Convolucional 1:

 Conv2D(32, (3,3), activation='relu', input_shape=(32,32,3))

 MaxPooling2D(pool_size=(2,2))

 Añadir el Bloque Convolucional 2:

 Conv2D(64, (3,3), activation='relu')

 MaxPooling2D(pool_size=(2,2))

 Añadir el Clasificador:

 Flatten()

 Dense(64, activation='relu')

 Dense(10, activation='softmax')

 Mostrar el resumen del modelo con model.summary().

 Documentar el código y explicar la función de cada capa.

 Responder la Pregunta de Análisis 1 en una celda Markdown del Notebook.

💬 Pregunta de Análisis 1

El documento “Áreas de aplicación” muestra un diagrama del procesamiento de visión artificial tradicional (página 9).
¿Qué partes de tu modelo de Keras corresponden a los pasos 1-3 ("Input, Preprocessing, Feature extraction") y qué parte corresponde al paso 4 ("Classifier")?
¿Cómo ha automatizado la CNN el proceso de "Feature extraction"?

🧩 Pistas para responder:

Input, Preprocessing, Feature extraction: capas Input, Conv2D, MaxPooling2D.

Classifier: capas Flatten, Dense, Output.

La CNN automatiza la extracción de características al aprender directamente los filtros y patrones de los datos, sin necesidad de definirlos manualmente.

✅ Criterios de finalización

El modelo se crea sin errores y su estructura coincide con la descripción.

El model.summary() muestra la secuencia correcta de capas.

La explicación de cada capa está documentada en el Notebook.

La Pregunta de Análisis 1 está respondida con claridad en Markdown.
